rpc/json.go

```diff
@@ -235,58 +263,38 @@ func (c *jsonCodec) ParseRequestArguments(argTypes []reflect.Type, params interf
 	}
 }
 
-func countArguments(args json.RawMessage) (int, error) {
-	var cnt []interface{}
-	if err := json.Unmarshal(args, &cnt); err != nil {
-		return -1, nil
+// parsePositionalArguments tries to parse the given args to an array of values with the given types.
+// It returns the parsed values or an error when the args could not be parsed. Missing optional arguments
+// are returned as reflect.Zero values.
+func parsePositionalArguments(args json.RawMessage, callbackArgs []reflect.Type) ([]reflect.Value, RPCError) {
+	var params []interface{}
```

Comment:
It would help prevent a few allocations if `params := make([]interface{}, 0, len(callbackArgs))`